pkg/minibroker/minibroker.go

```diff
@@ -325,13 +329,7 @@ func (c *Client) Provision(instanceID, serviceID, planID, namespace string, acce
 			return "", errors.Wrapf(err, "Failed to set operation key when provisioning instance %s", instanceID)
 		}
 		go func() {
-			err = c.provisionSynchronously(instanceID, namespace, serviceID, planID, chartName, chartVersion, provisionParams)
-			if err == nil {
-				err = c.updateConfigMap(instanceID, map[string]interface{}{
-					OperationStateKey:       string(osb.StateSucceeded),
-					OperationDescriptionKey: fmt.Sprintf("service instance %q provisioned", instanceID),
-				})
-			} else {
+			fail := func(err error) {
```

Comment:
Hmm, this seems a bit awkward. Basically why I had the silly `provisionSynchronously()` wrapper, but I can see how that might look silly.

Reply:
`provisionSynchronously` was taking too much responsibility, requiring too many parameters to be passed to it, making it confusing. This seems a bit awkward indeed, but more clear.